core/Azure.Mcp.Core/src/Services/Telemetry/TelemetryService.cs

```diff
@@ -106,6 +106,54 @@ public TelemetryService(IMachineInformationProvider informationProvider,
         return activity;
     }
 
+    /// <summary>
+    /// <inheritdoc/>
+    /// </summary>
+    public Activity? StartActivity<T>(string activityName, RequestContext<T> request) where T : RequestParams
+    {
+        if (!_isEnabled)
+        {
+            return null;
+        }
+
+        CheckInitialization();
+
+        var activity = Parent.StartActivity(activityName);
```

Comment:
to decrease duplicate logic.. can this call the parent method above

Reply:
Done